test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Computer.Tests.ps1

```diff
@@ -0,0 +1,97 @@
+# note these will manipulate private data in the PowerShell engine which will
+# enable us to not actually rename the system, but return right before we do
+$TesthooksType = [system.management.automation.internal.internaltesthooks]
+$TesthookName = "TestRenameComputer"
+$TesthookResultName = "TestRenameComputerResults"
+$DefaultResultValue = 0
+function Set-HookResult([int]$result) {
+    ${TesthooksType}::SetTestHook($TesthookResultName, $result)
+}
+
+# protect the user from from stop their computer 
+function Test-TesthookIsSet() {
+    try {
+        return ${TesthooksType}.GetField($TesthookName, "NonPublic,Static").GetValue($null)
+    }
+    catch {
+        # fall through
+    }
+    return $false

```

Comment:
why not have this in the catch?

Reply:
cause I need to return false from here anyway - no reason to have
```
catch {
    return $false
}
return $false
```